echemdb/cv/cv_entry.py

```diff
@@ -0,0 +1,262 @@
+        Additional keyword arguments are passed to the data frame plotting
+        method::
+
+            >>> entry.thumbnail(width=4, height=2, color='red', linewidth=2)
+            b"\x89PNG..."
+
+        """
+        kwds.setdefault("color", "b")
+        kwds.setdefault("linewidth", 1)
+        kwds.setdefault("legend", False)
+
+        import matplotlib.pyplot
+
+        # A reasonable DPI setting that should work for most screens is the default value of 72.
+        fig, axis = matplotlib.pyplot.subplots(
+            1, 1, figsize=[width / dpi, height / dpi], dpi=dpi
+        )
+        self.df.plot(
+            "E",
+            self._normalize_field_name("j"),
+            ax=axis,
+            **kwds,
+        )
+
+        matplotlib.pyplot.axis("off")
+        matplotlib.pyplot.close(fig)
+
+        import io
+
+        buffer = io.BytesIO()
+        fig.savefig(buffer, format="png", transparent=True, dpi=dpi)
+
+        buffer.seek(0)
+        return buffer.read()
+
+    def plot(self, x_label="E", y_label="j"):
+        r"""
+        Return a plot of this entry.
+        The default plot is a cyclic voltammogram ('j vs E').
+        When `j` is not defined `I` is used instead.
+
+        EXAMPLES::
+
+            >>> entry = CVEntry.create_examples()[0]
+            >>> entry.plot()
+            Figure(...)
+
+        The plot can also be returned with custom axis units available in the resource::
+
+            >>> entry.plot(x_label='t', y_label='E')
+            Figure(...)
+
+        The plot with axis units of the original figure can be obtained by first rescaling the entry::
+
+            >>> rescaled_entry = entry.rescale('original')
+            >>> rescaled_entry.plot()
+            Figure(...)
+
+        """
+        import plotly.graph_objects
+
+        x_label = self._normalize_field_name(x_label)
+        y_label = self._normalize_field_name(y_label)
+
+        fig = plotly.graph_objects.Figure()
+
+        def figure_name():
+            try:
+                self.source.figure
+            except AttributeError:
+                try:
+                    self.source.curve
+                except AttributeError:
+                    return self.identifier
+            return f"Fig. {self.source.figure}: {self.source.curve}"
+
+        # pylint: disable=duplicate-code
+        fig.add_trace(
+            plotly.graph_objects.Scatter(
+                x=self.df[x_label],
+                y=self.df[y_label],
+                mode="lines",
+                name=f"{figure_name()}",
+            )
+        )
+
+        def reference(label):
+            if label == "E":
+                try:
+                    self.package.get_resource("echemdb").schema.get_field(label)[
+                        "reference"
+                    ]
+                except KeyError:
+                    return ""
+                return f" vs. {self.package.get_resource('echemdb').schema.get_field(label)['reference']}"
```

Comment:
```suggestion
                field = self.package.get_resource("echemdb").schema.get_field(label)
                if "reference" not in field:
                    return ""
                return f" vs. {field["reference"]}"
```

While some consider this Pythonic, I really dislike exceptions for expected control flow.